app/services/jwt_token.rb

```diff
@@ -3,17 +3,22 @@
 
 class JwtToken < Struct.new(:token)
   class << self
-    def encode(user, secret)
-      payload = prepare_payload(user, secret)
+    def encode(user, secret, expiration = nil)
+      payload = prepare_payload(user, secret, expiration)
       new JWT.encode(payload, secret)
     end
 
     private
 
-    def prepare_payload(user, secret)
+    def prepare_payload(user, secret, expiration)
       jti_raw = [secret, iat].join(':')
       jti = Digest::SHA256.hexdigest(jti_raw)
-      { user_id: user.id, iat: iat, jti: jti }
+      payload = {
+        user_id: user.id,
+        iat: iat,
+        jti: jti,
+      }
+      expiration ? payload.merge(exp: Time.now.to_i + expiration) : payload
```

Comment:
I think this is easier to read (and slightly more efficient since it doesn't have to copy the hash)
```suggestion
      payload[:exp] = Time.now.to_i + expiration if expiration
      payload
```